app/role/members/route.js

```diff
@@ -20,8 +20,29 @@ export default Ember.Route.extend({
     controller.set('currentUserRoles', model.currentUserRoles);
   },
 
+  redirect(model) {
+    let org = model.role.get('organization');
+    let currentUser = this.session.get('currentUser');
+    let currentUserMembership = model.memberships.filterBy('user.id', currentUser.get('id'))[0];
+
+    if ((model.role.get('isPlatform') && currentUser.canManagePlatform(model.currentUserRoles, org)) ||
+        (model.role.get('isCompliance') && currentUser.canManageCompliance(model.currentUserRoles, org)) ||
+        (currentUserMembership && currentUserMembership.get('privileged'))) {
```

Comment:
Can this giant conditional be converted into a function that's a little easier to read?